controllers/nodehealthcheck_controller_test.go

```diff
@@ -556,32 +594,48 @@ func newNodeHealthCheck() *v1alpha1.NodeHealthCheck {
 }
 
 func newNodes(unhealthy int, healthy int) []runtime.Object {
```

Comment:
Nit: I suggest removing `newNodes` method. either modifying `isControlPlane` in `newNodesWithRole` to an optional arg  or keep it as is, for always  have an explicit role

Reply:
I was a bit too lazy for refactoring... done 😉